api/admin/controller/collection_settings.py

```diff
@@ -315,10 +320,14 @@ def process_settings(self, settings, collection):
             collection.protocol,
             is_child=(flask.request.form.get("parent_id") is not None),
         )
+        if isinstance(settings_class, ProblemDetail):
+            return settings_class
+        if settings_class is None:
+            return PROTOCOL_DOES_NOT_SUPPORT_SETTINGS
         collection_settings = {}
         for setting in settings:
-            key = setting.get("key")
-            value = flask.request.form.get(key)
+            key = setting["key"]
+            value = self._extract_form_setting_value(setting, flask.request.form)
             if key == "external_account_id":
```

Comment:
@RishiDiwanTT A question about the `external_account_id`: it is a field on `ODLSettings`/`ODL2Settings`, but is being set only on the collection, not the integration settings. Is that the intention?